### DISCUSSION TITLE
"Manually" ending a trial (and store trial data)

### DISCUSSION THREAD

**Discussion by psyxcience** [jspsych/jsPsych#2947] [Q&A] [answered]
I want to use a joystick as input (stimulus categorization by pushing and pulling). As I have not found a plugin, I use a normal html-keyboard response but include the following code in the "on_load" function:

```
    on_load: function() {
      intervalID = setInterval(function(){
          const myGamepad = navigator.getGamepads()[0];
          if ( myGamepad.axes[1] > 0.1 ) { 
              console.log("pull");
              jsPsych.finishTrial();
             };
          if ( myGamepad.axes[1] < -0.1 ) {
              console.log("push")
              jsPsych.finishTrial( );
             };          
        }, 100)
      },
```

This seems to work: It checks every 100 ms for the joystick position and interrupts the trial (correct 'push' and 'pull' console output). However, it does not record/store any data. I guess with using "jsPsych.finishTrial()" I jump that part specified in the plugin itself. Is there any way to get this to work?

I tried with including information I found in the plugin like
```
jsPsych.finishTrial( { rt: response.rt } )
```
But I guess that I cannot access "response" from outside the plugin(?)

I appreciate any suggestions.
Thanks,
Torsten

**Comment by jodeleeuw** [comment, marked as answer]
Hi @psyxcience,

The `finishTrial()` method isn't really intended to be called outside the plugin context. You can do it, but it is likely to create problems.

I would recommend modifying the plugin and adding joystick input directly to the plugin code.

Don't forget to cancel the interval function once you are done with the trial.

**Reply by psyxcience** [reply]
:( I was afraid that would be the answer. I'll try it.
Thanks a lot for the quick response!
(and for jsPsych!)
